src/data/real/nnreal.lean

```diff
@@ -228,6 +228,10 @@ instance : canonically_ordered_add_monoid ℝ≥0 :=
   ..nnreal.order_bot,
   ..nnreal.linear_order }
 
+instance : canonically_linear_ordered_add_monoid ℝ≥0 :=
```

Comment:
now you have both `canonically_linear_ordered_add_monoid` and `canonically_ordered_add_monoid` for nnreal, but the second one can be deduced from the linear one. I don't know exactly why and somebody else could comment on that, but if I remember correctly it is better to register only `canonically_linear_ordered_add_monoid` (for which you would just merge both proofs).

Reply:
The non-linear one is used to construct the linear one, so I can't not have the non-linear one registered, as far as I understand. I don't think there is an issue of having both, since the linear one is constructed via a `{ .. instance }`. But I could b wrong.

Reply:
In fact, the current proof of `canonically_ordered_add_monoid` is sufficient to prove the stronger linear instance. You can write:
```
instance : canonically_linear_ordered_add_monoid ℝ≥0 :=
{ add_le_add_left       := assume a b h c, @add_le_add_left ℝ _ a b h c,
  lt_of_add_lt_add_left := assume a b c, @lt_of_add_lt_add_left ℝ _ a b c,
  le_iff_exists_add     := assume ⟨a, ha⟩ ⟨b, hb⟩,
    iff.intro
      (assume h : a ≤ b,
        ⟨⟨b - a, le_sub_iff_add_le.2 $ by simp [h]⟩,
          nnreal.eq $ show b = a + (b - a), by rw [add_sub_cancel'_right]⟩)
      (assume ⟨⟨c, hc⟩, eq⟩, eq.symm ▸ show a ≤ a + c, from (le_add_iff_nonneg_right a).2 hc),
  ..nnreal.comm_semiring,
  ..nnreal.order_bot,
  ..nnreal.linear_order }
```
That is, you just add linear in the name.

Reply:
> but if I remember correctly it is better to register only `canonically_linear_ordered_add_monoid` (for which you would just merge both proofs).

I don't think it makes any difference - in fact, for `nat` we register them all separately due to claims that it makes typeclass lookup more performant.

However, if the two proofs are easy to merge, saving line count seems like a good enough reason to me!

Reply:
Feel free to commit the change or leave a suggestion that can be committed.

Reply:
Looks like the change I committed broke it...